lua/nvim-treesitter-playground/query_linter.lua

```diff
@@ -38,28 +37,17 @@ local function table_contains(predicate, table)
   return false
 end
 
-local function query_lang_from_playground_buf(buf)
-  for lang_buf, entry in pairs(playground_module._entries or {}) do
-    if entry.query_bufnr == buf then
-      return parsers.get_buf_lang(lang_buf)
-    end
-  end
-end
-
-function M.lint(buf)
+function M.lint(buf, lang_tree)
   buf = buf or api.nvim_get_current_buf()
   M.clear_virtual_text(buf)
   M.lints[buf] = {}
 
-  local filename = api.nvim_buf_get_name(buf)
-  local ok, query_lang = pcall(vim.fn.fnamemodify, filename, ":p:h:t")
-  query_lang = filename ~= "" and query_lang
-  local query_lang = ok and query_lang
-  if not query_lang then
-    query_lang = query_lang_from_playground_buf(buf)
-  end
+  lang_tree = lang_tree or parsers.get_parser(buf)
```

Comment:
@steelsojka  i don't think this will work. `parsers.get_parser(buf)` will yield `query` because we only lint on query buffers. We need the location of the file to find out for want language the queries are.

Reply:
Yeah I see the problem now. I think I need to revert most of the changes in this file and only add a couple lines and it should work. I don't want to drop the feature or anything.

Reply:
Take a look now and see if that would work better.